pkg/util/strategicpatch/patch.go

```diff
@@ -240,15 +602,20 @@ func mergeSlice(original, patch []interface{}, elemType reflect.Type, mergeKey s
 	return original, nil
 }
 
-// This panics if any element of the slice is not a map.
+// This method no longer panics if any element of the slice is not a map.
```

Comment:
Okay, in that case I would recommend extending the function to return an error. Otherwise it may create some hard-to-find bugs if this error case is hidden, since the function would act like the value is not found when in fact it may be encountering an element of the wrong type.

I did think that panics were generally logged and recovered in our daemons, but not sure if we want to depend on that kind of handling. @bgrant0607 is there an expectation for how severely panics can affect daemons like apiserver that can be depended on?